### ISSUE TITLE
Metabase can't find Snowflake tables

### ISSUE THREAD

**Issue by leonardomatsuura** [metabase/metabase#8864]
Hello,

Metabase doesn't seem to be able to sync with tables when connecting to a Snowflake database:
![no_tables](https://user-images.githubusercontent.com/38956920/48085900-ae6c6d00-e1e2-11e8-8372-07ce4142ae9c.png) even tough i can see them on the Snowflake console (table names hidden):
<img width="1440" alt="snowflake_console" src="https://user-images.githubusercontent.com/38956920/48086295-aa8d1a80-e1e3-11e8-81b2-0ee89fd0fd35.png">

Here's my setup:

- Your operating system: macOS
- Your databases: Snowflake
- Metabase version: [v0.31.0-RC1](https://github.com/metabase/metabase/releases/tag/v0.31.0-RC1)
- Metabase hosting environment: java -jar
- Metabase internal database: H2
- Snowflake connection (account and username hidden): 
![connection](https://user-images.githubusercontent.com/38956920/48086059-1327c780-e1e3-11e8-98df-9b529e8f1ca7.png)
- Metabase logs: [metabase_logs.txt](https://github.com/metabase/metabase/files/2554565/metabase_logs.txt)

*Repeatable steps to reproduce the issue*
- [Download the jar]((https://github.com/metabase/metabase/releases/tag/v0.31.0-RC1))
- `java -jar metabase.jar` and setup a Snowflake connection
- Trigger a database sync
- Try to build query through query builder
-------

**Comment by salsakran** [comment]
Are the tables present in the admin panel, under the Data Model section?

Also do they show up on a hard refresh or do they just never show up? It looks like the sync process found something, trying to narrow down whether we created entries for them and there's a bug displaying them or whether they're not being stored.

Also, is there anything in the javascript logs?

**Comment by leonardomatsuura** [comment]
> Are the tables present in the admin panel, under the Data Model section?

No

>Also do they show up on a hard refresh or do they just never show up? It looks like the sync process found something, trying to narrow down whether we created entries for them and there's a bug displaying them or whether they're not being stored.

What is a hard refresh? Is it clicking on `Sync Database Schema Now` in the connection tab? I've done that and it doesn't work.

>Also, is there anything in the javascript logs?

Where are the javascript logs located? I'm using Chrome and I've opened the console and all that it shows is: 
![image](https://user-images.githubusercontent.com/38956920/48090622-91d63200-e1ee-11e8-9b35-9fae807f2b57.png)

Thanks for the fast reply!

**Comment by camsaul** [comment]
Hi @leonardomatsuura, it looks like the database is `ORIGIN` in your console but you've specified `XLARGE_LOAD_DB` in the connection settings.

**Comment by leonardomatsuura** [comment]
Hi @camsaul thanks for pointing that out. My screenshot was bad and didn't really show what I meant. I've reupload a new one, where you can see `XLARGE_LOAD_DB` is the database and `ORIGIN` is a schema.
<img width="1434" alt="snowflake_console" src="https://user-images.githubusercontent.com/38956920/48130189-faff8900-e272-11e8-8739-0d18de33d327.png">
I've only changed the screenshot, the setup is still the same

**Comment by leonardomatsuura** [comment]
I was able to find a workaround on this issue. I noticed this query running when i forced a database sync:
`show /* JDBC:DatabaseMetaData.getTables() */ tables in schema "Snowflake"."ORIGIN";` and Snowflake was the name I set on the connection. When setting the connection name to the database all seemed to start working and I can see the tables:
![image](https://user-images.githubusercontent.com/38956920/48152189-5436df00-e2aa-11e8-9fc1-801a5211eb1f.png)

Apparently metabase is using the Name we use to identify a Database instead of the actual database when searching for tables https://github.com/metabase/metabase/blob/master/src/metabase/driver/snowflake.clj#L171

**Comment by camsaul** [comment]
oh yeah that's a bug. Good catch. I will fix it

**Comment by johnnykarate321** [comment]
Hi Guys. I believe I've run into a similar issue but not sure it's related to this bug. I've setup a SnowFlake data source which saves correctly but when the system goes into "Exploring" mode it will run forever.  (I've upgraded the Metabase Elastic Beanstalk Environment to the latest and greatest V31.1)
Maybe this is as designed but the database name is case sensitive in Snowflake, so the query is failing at the database layer but the "Exploring" mode doesn't error out. I've modified the database name to be all UPPER case and now everything is working properly. (Also is running way faster as compared to using a Redshift database) Attached is the query that was dying when the "Exploring" mode was still running in a FireFox V63.0.3 browser window.
![metabasesnowflake](https://user-images.githubusercontent.com/43764034/49108567-5ce15c00-f23d-11e8-8080-c6864809b7c6.png)
